src/Raven.Server/Documents/Handlers/Processors/Attachments/AbstractAttachmentHandlerProcessorForGetAttachment.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Raven.Client;
+using Raven.Client.Documents.Attachments;
+using Raven.Client.Documents.Operations.Attachments;
+using Raven.Server.ServerWide;
+using Raven.Server.Web;
+using Sparrow.Json;
+using Sparrow.Logging;
+
+namespace Raven.Server.Documents.Handlers.Processors.Attachments
+{
+    internal abstract class AbstractAttachmentHandlerProcessorForGetAttachment<TRequestHandler, TOperationContext> : AbstractHandlerProcessor<TRequestHandler, TOperationContext>
+        where TRequestHandler : RequestHandler
+        where TOperationContext : JsonOperationContext
+    {
+        protected Logger Logger;
+        private readonly bool _isDocument;
+
+        protected AbstractAttachmentHandlerProcessorForGetAttachment([NotNull] TRequestHandler requestHandler, [NotNull] JsonContextPoolBase<TOperationContext> contextPool, Logger logger, bool isDocument) : base(requestHandler, contextPool)
+        {
+            Logger = logger;
+            _isDocument = isDocument;
+        }
+
+        protected abstract ValueTask<AttachmentResult> GetAttachmentAsync(TOperationContext context, string documentId, string name, AttachmentType type, string changeVector);
+
+        protected abstract CancellationToken GetDataBaseShutDownToken();
```

Comment:
Why not use RequestHandler.GetOperationToken?

Reply:
done